src/main/java/org/cryptomator/cryptofs/health/shortened/ShortenedNamesCheck.java

```diff
@@ -103,6 +115,36 @@ void checkShortenedName(Path dir) throws IOException {
 			}
 		}
 
+
+		/**
+		 * Checks if the string stored inside the name file is base64url encoded
+		 *
+		 * <em>visible for testing</em>
+		 *
+		 * @return {@link NameEncoding} indicating if it is valid, invalid or affected by https://github.com/cryptomator/cryptofs/issues/121
+		 */
+		NameEncoding isValidEncodedName(String toAnalyse) {
+			int firstNullByte = toAnalyse.indexOf('\0');
+			var beforeFirstNull = firstNullByte == -1 ? toAnalyse : toAnalyse.substring(0, firstNullByte);
+			var afterFirstNull = firstNullByte == -1 ? "" : toAnalyse.substring(firstNullByte);
+
+			boolean isFirstBase64 = BASE64URL.canDecode(beforeFirstNull);
+
+			if (isFirstBase64 && afterFirstNull.isEmpty()) {
+				return NameEncoding.VALID;
+			} else if (isFirstBase64 && NULL_BYTES.matcher(afterFirstNull).matches()) {
+				return NameEncoding.BUG121;
+			} else {
+				return NameEncoding.INVALID;
+			}
+		}
```

Comment:
I think cutting the string in halves at the first occurrence of a null char is overly complicated.

```suggestion
		NameEncoding isValidEncodedName(String toAnalyse) {
			var trimmed = toAnalyse.trim();
			return !toAnalyse.equals(trimmed) //
				? NameEncoding.BUG121 // string padded with chars < 0x20
				: BASE64URL.canDecode(toAnalyse) ? NameEncoding.VALID : NameEncoding.INVALID;
		}
```

Reply:
My goal was to be as precise as possible: Only return `NameEncoding.BUG121`, if the string is base64url encoded and has trailing null bytes.

But we can widen the scope of the check a little, we should be only careful in doing so and make as less assumptions as possible. For the suggestion the assumption is not only "affected by the bug" but also "it might happen, that chars < 0x20 are added before and after the targeted string".

Reply:
In general: We should keep the check, the result and its fix as narrow scoped as possible, to not fix an error in the wrong way.

There can be a lot of things broken in the vault structure, but in my opinion better stick to only the things with known causes or unambigous repairs. In the suggestion, the cause is unknown _and_ the repair is ambigous, because the string between the padding can be still invalid base64url.

Reply:
After a short discussion we agreed to the following changes (as implemented in db83548 ): Make a basic syntax check by searching for `.c9r`in the long name and check for trailing bytes after and that the suffix is base64url encoded.